modules/starfish/2.0/starfish2.smk

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Ryan Morin
+# Module Author:    Laura Hilton
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+import glob 
+import itertools as it
+import string
+import copy
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["starfish"]`
+CFG = op.setup_module(
+    name = "starfish",
+    version = "2.0",
+    subdirectories = ["inputs", "starfish", "outputs"],
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _starfish_input_vcf,
+    _starfish_rename_output, 
+    _starfish_output_vcf,
+    _starfish_output_union,
+    _starfish_output_venn,
+    _starfish_all,
+
+##### GLOBAL VARIABLES #####
+
+# All variant callers from config
+callers = list(CFG["inputs"]["vcf"].keys())
+callers = [caller.lower() for caller in callers]
+
+# The name of the union VCF file
+union_vcf = copy.deepcopy(callers)
```

Comment:
Wouldn't it be easier to just call .keys twice? If you're worried about getting them in the same order I doubt that's a problem. You don't have to change it. Just thought I would point this out.

Reply:
If I call `.keys()` again I also have to call the `.lower` function again, so I think this is more efficient purely for that reason. 

But seriously one of the most annoying things about Python is that you have to explicitly copy a list to a new variable.

Reply:
Perl is so much better for this kind of crap. It's a lazy language for lazy coders. 
Is this call to "lower" because you don't trust the user to use the lowercase name? Isn't your config already lowercase? I must be missing something here.

Reply:
I don't want to assume the user has used all lowercase letters in the keys in the config. Since my function to replace the generic Starfish output vcf names with tool names depends on case-sensitive find-and-replace, I wanted to guarantee that there were no upper case letters in these keys that might inadvertently get replaced and garble the output file names.